src/bgfx.cpp

```diff
@@ -1291,10 +1291,10 @@ namespace bgfx
 		m_frame->m_renderItem[renderItemIdx].draw = m_draw;
 		m_frame->m_renderItemBind[renderItemIdx]  = m_bind;
 
-		if (!_preserveState)
+		m_draw.clear(_flags);
+		m_bind.clear(_flags);
+		if (_flags == BGFX_DISCARD_ALL)
```

Comment:
This seems to not be following way how discard flags work with `bgfx::discard`

Reply:
```
m_uniformBegin = m_uniformEnd;
```
only for BGFX_DISCARD_STATE?